packages/did-mailto/src/index.js

```diff
@@ -0,0 +1,70 @@
+export const foo = 1
+
+/**
+ * create a did:mailto from an email address
+ *
+ * @param {import("./types").EmailAddress} email
+ * @returns {import("./types").DidMailto}
+ */
+export function fromEmail(email) {
+  const { domain, local } = parseEmail(email)
+  const did = /** @type {const} */ (
+    `did:mailto:${encodeURIComponent(domain)}:${encodeURIComponent(local)}`
+  )
+  return did
+}
+
+/**
+ * @param {string} email
+ */
+function parseEmail(email) {
+  const atParts = email.split('@')
+  if (atParts.length < 2) {
+    throw new TypeError(
+      `expected at least 2 @-delimtied segments, but got ${atParts.length}`
+    )
+  }
+  const domain = atParts.at(-1) ?? ''
```

Comment:
Empty domain does not seems valid and should be an error perhaps worth considering checking it also has `.` in it.

Reply:
Nit: I personally would find it easier to follow this if it was something like:

```ts
const offset = email.lastIndexOf('@')
if (offset < 0 || email.length - offset <= 1) {
   throw
} else {
  const local = emal.slice(0, offset)
  const domain = email.slice(offset + 1)
}
```